[skjson] Rename sk_freeze and remove unused export

bennostein opened the pull request:
I noticed while adding an optional default parameter to Values#getUnique that it's probably a footgun to export `sk_freeze` from skjson via `core`'s index.ts, since it doesn't actually _freeze_, just adds the hidden property that we use to track frozen-ness.

This PR drops that export and renames to avoid confusion.

jberdine commented on skipruntime-ts/core/src/index.ts:
```diff
@@ -54,7 +54,7 @@ import {
 } from "./binding.js";
 
-export { sk_freeze, isSkManaged };
+export { deepFreeze, isSkManaged };
```
deepFreeze is already exported to the public api from api.ts

jberdine reviewed (approved):
Agree that the `sk_freeze` name is not ideal. Made a suggestion for a maybe better one.

Note though that core/src/index.ts is the internal interface of the core package, the one we expect users to use is only the part in api.ts.

bennostein commented:
> Agree that the `sk_freeze` name is not ideal. Made a suggestion for a maybe better one.

Suggestion taken!
 
> Note though that core/src/index.ts is the internal interface of the core package, the one we expect users to use is only the part in api.ts.

Oh right, of course!  So I guess this is just taking the footgun away from _us_, not users.  Renamed the PR and updated the description, thanks Josh

jberdine commented:
> > Note though that core/src/index.ts is the internal interface of the core package, the one we expect users to use is only the part in api.ts.
> 
> Oh right, of course! So I guess this is just taking the footgun away from _us_, not users. Renamed the PR and updated the description, thanks Josh

Well, I don't know that there is anything that actually prevents users from importing this stuff, it's just not documented.
